### PR TITLE
Mustang/1051 user profile

### PR DESCRIPTION
**Pull request by jasonmule**
This pull request makes the following changes:
- Moves user profile settings to a modal

Test these changes by:
- Click user settings to make the modal visible and navigate to user profile settings or notification settings.

Fixes ushahidi/platform#1051 and closes ushahidi/platform#1052 .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/195)

### CODE REVIEW

File: app/user-profile/directives/notifications.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                var contactHasChanged = function (contact) {
+                    return contact.original !== contact.contact;
+                };
+
+                loadContacts();
+                loadNotifications();
+
+                $scope.deleteNotification = function (notification) {
+                    $translate('notify.notification.delete_confirm').then(function (message) {
+                        Notify.showConfirm(message).then(function () {
```
Comment (jasonmule) [inline app/user-profile/directives/notifications.js]: use showConfirmAlert

File: app/user-profile/directives/notifications.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                            }, function () {
+                                showErrorMessage('notification.error_message');
+                            });
+                        });
+                    });
+                };
+
+                $scope.deleteContact = function (contact) {
+                    $translate('notify.contact.delete_confirm').then(function (message) {
+                        Notify.showConfirm(message).then(function () {
```
Comment (jasonmule) [inline app/user-profile/directives/notifications.js]: use showConfirmAlert

**Comment by jasonmule** [comment]
@rjmackay @willdoran This is more or less done. 2 issues that I didn't troubleshoot yet:
- Animation/ fade-in of the modal is not quite right.
- Scroll bars on the notification settings do not appear when listings don't fit the view.

**Comment by rjmackay** [comment]
This is looking pretty good.

> Scroll bars on the notification settings do not appear when listings don't fit the view.

I think I left a `todo` note about this in the modal code - the PL has some JS that limits the modal height based on screen size - we need to replicate that without the jquery



---

Reviewed 20 of 20 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[app/user-profile/directives/account-settings.js, line 17 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KIPio9k85fZ2mz7Ontw:-KIPio9k85fZ2mz7Ontx:-652809487) ([raw file](https://github.com/ushahidi/platform-client/blob/e92f5860edc7d11fd87719a049fc1012c7d3d12a/app/user-profile/directives/account-settings.js#L17)):*
> ```JavaScript
>         templateUrl: 'templates/users/account_settings.html',
>         link: function ($scope) {
>             $rootScope.$on('event:show:account_settings', function () {
> ```

:+1: I like how this make the directive self contained!

---

*[app/user-profile/directives/notifications.js, line 105 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KIHArGOo0BM1LA6dQJM-r1-105:-KIPjsw3uvVh20FeQjlF:467340616) ([raw file](https://github.com/ushahidi/platform-client/blob/e92f5860edc7d11fd87719a049fc1012c7d3d12a/app/user-profile/directives/notifications.js#L105)):*
<details><summary><i>Previously, jasonmule (Jason Mule) wrote…</i></summary>
> use showConfirmAlert
</details>
if you try this in the PL - it uses a slider at the top of the modal. 
https://www.dropbox.com/s/s9o0mqvu6op3767/Screenshot%202016-05-23%2011.15.14.png?dl=0

I don't think we have time to switch from a delete confirm, to 'undo' right now. But ideally we should use a slider inside the modal for the confirmation since we already have a modal open.

---

*[app/user-profile/directives/notifications.js, line 125 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KIHArGOo0BM1LA6dQJM-r1-125:-KIPkRci9VGYWuNRpgmh:-1305279109) ([raw file](https://github.com/ushahidi/platform-client/blob/e92f5860edc7d11fd87719a049fc1012c7d3d12a/app/user-profile/directives/notifications.js#L125)):*
<details><summary><i>Previously, jasonmule (Jason Mule) wrote…</i></summary>
> use showConfirmAlert
</details>
as above: ideally we should use a slider inside the modal for the confirmation since we already have a modal open.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-:-KIPjZVXNUjsiRtk36jN:-1242075621)*

**Comment by rjmackay** [comment]
It looks like there was some confusion over how ModalService was intended to be used - the event here are making things more complex and coupled rather than less.. suggested changes below.

<details><summary><i>Previously, coveralls wrote…</i></summary>
> 
> [Coverage Status](https://coveralls.io/builds/6426517)
> 
> Coverage decreased (-0.4%) to 56.24% when pulling **45b194a6a35560395f1271091f6df87363405abc on mustang/1051-user-profile** into **f35e0ff670b87216afa527795e9482334ed7b4e4 on develop**.
> 
</details>


---

Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[app/common/directives/modal-container.directive.js, line 9 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KJDzKed2JUsN1JkMndI:-KJDzKed2JUsN1JkMndJ:-1474237891) ([raw file](https://github.com/ushahidi/platform-client/blob/45b194a6a35560395f1271091f6df87363405abc/app/common/directives/modal-container.directive.js#L9)):*
> ```JavaScript
> ModalContainer.$inject = ['$timeout', '$rootScope', '$compile', 'ModalService'];
> function ModalContainer($timeout, $rootScope, $compile, ModalService) {
>     var scopeToDestroy;
> ```

Can you just call this `templateScope` and keep it consistent everywhere?
you'll need to rename the param in `openModal` to scopeArg or something but thats ok I think.

---

*[app/common/directives/modal-container.directive.js, line 42 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KJDzZ2NrEggo15VuzSQ:-KJDzZ2NrEggo15VuzSR:-1095254517) ([raw file](https://github.com/ushahidi/platform-client/blob/45b194a6a35560395f1271091f6df87363405abc/app/common/directives/modal-container.directive.js#L42)):*
> ```JavaScript
>         function openModal(ev, template, title, icon, templateScope, closeOnOverlayClick, showCloseButton) {
> 
>             templateScope = templateScope || $scope.$new();
> ```

I think this needs to become 
`templateScope = templateScope ? templateScope.$new || $scope.$new();`
to ensure we always have a scope dedicated to our modal - otherwise when we called destroy it might destroy a scope thats still used elsewhere

---

*[app/common/directives/modal-container.directive.js, line 84 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KJDzs8k959S0_OxHNGs:-KJDzs8k959S0_OxHNGt:-1596134847) ([raw file](https://github.com/ushahidi/platform-client/blob/45b194a6a35560395f1271091f6df87363405abc/app/common/directives/modal-container.directive.js#L84)):*
> ```JavaScript
>             $scope.classVisible = false;
>             $rootScope.toggleModalVisible();
>             scopeToDestroy.$destroy();
> ```

To be complete call `modalContent.html('');` too so we empty the HTML from the DOM

---

*[app/frame/directives/mode-bar.js, line 22 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KJE-moB1m_XvBpH7T70:-KJE-moB1m_XvBpH7T71:1688735534) ([raw file](https://github.com/ushahidi/platform-client/blob/45b194a6a35560395f1271091f6df87363405abc/app/frame/directives/mode-bar.js#L22)):*
> ```JavaScript
>             // Show account settings modal
>             $scope.viewAccountSettings = function () {
>                 $rootScope.$emit('event:show:account_settings');
> ```

This should be replaced by a call to `ModalService.openTemplate`

---

*[app/user-profile/directives/account-settings.js, line 21 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KJE-HSRj2R_x-iemjiI:-KJE-HSRj2R_x-iemjiJ:-4781322) ([raw file](https://github.com/ushahidi/platform-client/blob/45b194a6a35560395f1271091f6df87363405abc/app/user-profile/directives/account-settings.js#L21)):*
> ```JavaScript
>                 template = '<account-settings ng-show="visible"></account-settings>',
> 
>                 unbind = $rootScope.$on('event:show:account_settings', function () {
> ```

This is counter to the way I intended the ModalService to be used.
This should move into the mode bar code in a simple ngClick handler - there's event binding is overkill that we only really needed when this directive was always in the DOM.

Since this entire directive should be compiled fresh each time the settings is opened you can just do the user loading directly in `link`

---

*[server/www/index.html, line 45 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-KJE06UW24LSKepdjZxa:-KJE06UX7Up6XMgVi7FE:506842725) ([raw file](https://github.com/ushahidi/platform-client/blob/45b194a6a35560395f1271091f6df87363405abc/server/www/index.html#L45)):*
> ```HTML
> 
>           <mode-bar></mode-bar>
>           <account-settings ng-show="visible"></account-settings>
> ```

If you're using ModalService this shouldn't be here anymore.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/195#-:-KJE0FAOr9Oh_mFiphhw:1214150266)*

**Pull request merged**